gh-139257: escape enumerator-like letters to support docutils 0.22

stefanor opened the pull request:
Escape things that look like Roman numerals. Docutils 0.22 includes a new roman-numeral interpreter that falls over some of our content. Escape anything that causes trouble.

Python itself isn't using docutils 0.22 yet for its doc generation, but some Linux distributions have updated to it.

An alternative to GH-139258

<!-- gh-issue-number: gh-139257 -->
* Issue: gh-139257
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--140031.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

stefanor committed:
gh-139257: docutils 0.22 support

Escape things that look like Roman numerals. Docutils 0.22 includes a
new roman-numeral interpreter that falls over some of our content.
Escape anything that causes trouble.

Python itself isn't using docutils 0.22 yet for its doc generation, but
some Linux distributions have updated to it.

picnixz reviewed (approved):
Well, if this change is actually sufficient it's good. But that means that we need to be careful when adding roman-like text. What I find annoying is the Q/A modifications, but I don't know how we could do otherwise.

stefanor committed:
Replace the monkey patch to disable roman lists with an exception

We don't have a single clean method to disable roman lists across
docutils versions, but we can blow up the bulid cleanly...

stefanor committed:
Use an explicitly named link

Docutils 0.22 does not allow referencing a link that was implicitly
named earlier in the document, it complains:

  ERROR: Duplicate target name, cannot be used as a unique reference:
  "basic authentication".

stefanor committed:
Render the FAQ local tables of contents for all media

Docutils 0.22 doesn't allow the contents directive to be used within an
only directive, it complains:

  ERROR: The "contents" directive may not be used within topics or body
  elements.
